src/lir/converter/expression.rs

```diff
@@ -0,0 +1,992 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use core::panic;
+
+use solang_parser::pt::Loc;
+
+use crate::codegen;
+use crate::lir::converter::Converter;
+use crate::lir::expressions::{BinaryOperator, Expression, Operand, UnaryOperator};
+use crate::lir::instructions::Instruction;
+use crate::lir::vartable::Vartable;
+use crate::sema::ast;
+
+impl Converter<'_> {
+    pub(crate) fn lower_expression(
```

Comment:
This seems to be a crucial part of it. I'd appreciate some doc comment (maybe even on the module level) explaining how the translation works on a high level.

Reply:
Hi @xermicus, I added doc comments to almost all public functions and wrote a high-level explanation of what the 'lower_expression' does.